Python/Product/PythonTools/PythonTools/OutliningTaggerProvider.cs

```diff
@@ -171,30 +168,19 @@ internal class OutliningTagger : ITagger<IOutliningRegionTag>, IPythonTextBuffer
                 }
             }
 
-            internal static TagSpan GetTagSpan(ITextSnapshot snapshot, int start, int end, int headerIndex = -1) {
+            internal static TagSpan GetTagSpan(SnapshotPoint start, SnapshotPoint end) {
                 TagSpan tagSpan = null;
+                var snapshot = start.Snapshot;
                 try {
-                    // if the user provided a -1, we should figure out the end of the first line
-                    if (headerIndex < 0) {
-                        headerIndex = snapshot.GetLineFromPosition(start).End.Position;
+                    SnapshotPoint hintEnd = end;
+                    if (start.GetContainingLine().LineNumber + 5 < hintEnd.GetContainingLine().LineNumber) {
```

Comment:
Why `5`?

Reply:
Ask the original author. I suspect it's to prevent unnecessarily large tooltips.